test/runner.py

```diff
@@ -294,7 +294,17 @@ def run_tests(options, suites):
   print('Test suites:')
   print([s[0] for s in suites])
   # Run the discovered tests
-  testRunner = unittest.TextTestRunner(verbosity=2, failfast=options.failfast)
+
+  try:
+    import xmlrunner # type: ignore
+    # Output fd must be left open until after testRunner.run() below
+    output = open('test-results.xml', 'wb')
+    testRunner = xmlrunner.XMLTestRunner(output=output, verbosity=2,
+                                         failfast=options.failfast)
+    print('Writing XML test output to ' + os.path.abspath(output.name))
+  except ImportError:
+    testRunner = unittest.TextTestRunner(verbosity=2, failfast=options.failfast)
```

Comment:
Its generally accepted that you have to run `pip install -r requirements-dev.txt` before you can run any tests.. so I don't think this needs to be conditional.